pkgs/by-name/co/construct/package.nix

```
@@ -0,0 +1,37 @@
+{ llvmPackages
+, stdenv
+, lib
+, fetchFromGitHub
+, boost
+}:
+
+stdenv.mkDerivation rec {
+  pname = "construct";
+  version = "master";
+
+  src = fetchFromGitHub {
+    owner = "Thomas-de-Bock";
+    repo = pname;
+    rev = "b109a7f67253364af2c405506a461935c6c887b6";
+    sha256 = "sha256-pOoqjtFkvMtof+YjV47JPu5tw/U6CU/MpL+C1Z1pgrE=";
+  };
+
```

In order to build this package on darwin the following is necessary:

```suggestion

  postPatch = lib.optionalString stdenv.isDarwin ''
    substituteInPlace Makefile \
        --replace g++ c++
  '';
```